ui/server/src/main/scala/pl/touk/nussknacker/ui/api/AppResources.scala

```diff
@@ -45,35 +46,53 @@ class AppResources(config: Config,
     }
   }
 
+  private def toEntity(healthCheckProcessResponse: HealthCheckProcessResponse): ResponseEntity = {
+    val printer = Printer.noSpaces.copy(dropNullValues = true)
+    HttpEntity(
+      ContentTypes.`application/json`,
+      printer.pretty(healthCheckProcessResponse.asJson)
+    )
+  }
+
   def securedRoute(implicit user: LoggedUser): Route =
     pathPrefix("app") {
       path("healthCheck") {
+        get {
+          complete {
+            HttpResponse(status = StatusCodes.OK, entity = toEntity(HealthCheckProcessResponse(OK)))
+          }
+        }
+      } ~ path("healthCheck" / "process" / "deployed") {
         get {
           complete {
             notRunningProcessesThatShouldRun.map[HttpResponse] { set =>
               if (set.isEmpty) {
-                HttpResponse(status = StatusCodes.OK)
+                val respEntity = toEntity(HealthCheckProcessResponse(OK))
+                HttpResponse(status = StatusCodes.OK, entity = respEntity)
               } else {
                 logger.warn(s"Processes not running: ${set.keys}")
                 logger.debug(s"Processes not running - more details: $set")
-                HttpResponse(status = StatusCodes.InternalServerError, entity = s"Deployed processes not running (probably failed): \n${set.keys.mkString(", ")}")
+                val respEntity = toEntity(HealthCheckProcessResponse(ERROR, Some("Deployed processes not running (probably failed)"), Some(set.keys.toSet)))
+                HttpResponse(status = StatusCodes.InternalServerError, entity = respEntity)
               }
             }.recover[HttpResponse] {
               case NonFatal(e) =>
                 logger.error("Failed to get statuses", e)
-                HttpResponse(status = StatusCodes.InternalServerError, entity = "Failed to retrieve job statuses")
+                val respEntity = toEntity(HealthCheckProcessResponse(ERROR, Some("Failed to retrieve job statuses")))
+                HttpResponse(status = StatusCodes.InternalServerError, entity = respEntity)
             }
           }
         }
-      } ~ path("sanityCheck")  {
+      } ~ path("healthCheck" / "process" / "validation")  {
```

Comment:
let's try to be a bit more consistent. Either:
- validation and deployment 
- validated and deployed?